File: frontend/lib/api/fundamentals.ts
import { apiFetch } from './client';

export interface FundamentalsRead {
  ticker: string;
  pe_ratio: number | null;
  pb_ratio: number | null;
  fcf_yield: number | null;
  operating_margin: number | null;
  dividend_yield: number | null;
  disclaimer: string;
}

export function getFundamentals(ticker: string): Promise<FundamentalsRead> {
  return apiFetch<FundamentalsRead>(`/api/v1/stocks/${ticker}/fundamentals`);
}

File: backend/interfaces/rest/schemas/fundamentals.py
"""Pydantic-Schema für Fundamentaldaten-Response."""

from __future__ import annotations

from pydantic import BaseModel


class FundamentalsResponse(BaseModel):
    """Response-Schema für GET /api/v1/stocks/{ticker}/fundamentals."""

    ticker: str
    pe_ratio: float | None
    pb_ratio: float | None
    eps_chf: float | None
    dividend_yield_pct: float | None
    fcf_yield: float | None = None
    operating_margin: float | None = None
    disclaimer: str
